libs/@blockprotocol/graph/src/types/subgraph/element-mappings.ts

```diff
@@ -0,0 +1,120 @@
+import { BaseUri, VersionedUri } from "@blockprotocol/type-system";
+
+import { Entity, EntityId, EntityRecordId } from "../entity";
+import {
+  DataTypeWithMetadata,
+  EntityTypeWithMetadata,
+  OntologyTypeRecordId,
+  PropertyTypeWithMetadata,
+} from "../ontology";
+import { EntityValidInterval } from "./edges";
+import { EntityIdAndTimestamp } from "./edges/outward-edge";
+import {
+  DataTypeVertex,
+  EntityTypeVertex,
+  EntityVertex,
+  EntityVertexId,
+  OntologyTypeVertexId,
+  PropertyTypeVertex,
+} from "./vertices";
+
+/**
+ * A utility type that maps various ways of identifying a single (or series of) element(s) of the graph to their
+ * associated types.
+ *
+ * Helpful when creating generic functions that operate over a {@link Subgraph}
+ */
+export type GraphElementIdentifiers<Temporal extends boolean> =
+  | {
+      identifier: VersionedUri | OntologyTypeVertexId | OntologyTypeRecordId;
+      element:
+        | DataTypeWithMetadata
+        | PropertyTypeWithMetadata
+        | EntityTypeWithMetadata;
+      vertex: DataTypeVertex | PropertyTypeVertex | EntityTypeVertex;
+    }
+  | {
+      identifier: BaseUri;
+      element:
+        | DataTypeWithMetadata[]
+        | PropertyTypeWithMetadata[]
+        | EntityTypeWithMetadata[];
+      vertex: DataTypeVertex[] | PropertyTypeVertex[] | EntityTypeVertex[];
+    }
+  | {
+      identifier: EntityIdAndTimestamp | EntityVertexId | EntityRecordId;
```

Comment:
That's how it's used, sure. But it's just what it says, it's an `EntityId` and a `Timestamp`. 

Assuming you were to query a `Subgraph` for that, you'd either get back a single revision (because revisions of an entity do not overlap), or you'd get back `undefined`.

Reply:
This was done